cli/src/extension_api.ts

```diff
@@ -1,6 +1,62 @@
 // deno-lint-ignore-file ban-types
 
 export class PhylumApi {
+  /**
+   * Send a request to the Phylum REST API.
+   *
+   * See https://api.staging.phylum.io/api/v0/swagger/index.html for available API endpoints.
+   *
+   * The `options` parameter matches the `options` parameter of the `fetch` function:
+   * https://developer.mozilla.org/en-US/docs/Web/API/fetch
+   */
+  static async fetch(
+    apiVersion: ApiVersion | string,
+    endpoint: string,
+    options?: RequestInit,
+  ): Promise<object> {
+    // Ensure header object is initialized.
+    const fetchOptions = options ?? {};
+    fetchOptions.headers = fetchOptions.headers ?? {};
+
+    // Set Authorization/Content-Type headers if they are missing.
+    if (fetchOptions.headers instanceof Headers) {
+      if (!fetchOptions.headers.has("Authorization")) {
+        const token = await PhylumApi.getAccessToken();
+        fetchOptions.headers.set("Authorization", `Bearer ${token}`);
+      }
+
+      if (fetchOptions.headers.has("Content-Type")) {
+        fetchOptions.headers.set("Content-Type", "application/json");
+      }
+    } else {
+      if (!fetchOptions.headers["Authorization"]) {
+        const token = await PhylumApi.getAccessToken();
+        fetchOptions.headers["Authorization"] = `Bearer ${token}`;
+      }
+
+      if (!fetchOptions.headers["Content-Type"]) {
+        fetchOptions.headers["Content-Type"] = "application/json";
+      }
+    }
+
+    // Set Content-Type header if it is missing.
```

Comment:
This fails `deno check` because it doesn't handle this annoying but valid way to specify headers:

```
{ headers: [["Accept", "*/*"], ["Content-Type", "application/json"]] }
```

You can greatly simplify the code if you just go ahead and create the `Headers` object. Then there is only one case to handle.

```suggestion
    const headers = new Headers(fetchOptions.headers);

    // Set Authorization header if it is missing.
    if (!headers.has("Authorization")) {
      const token = await PhylumApi.getAccessToken();
      headers.set("Authorization", `Bearer ${token}`);
    }

    // Set Content-Type header if it is missing.
    if (headers.has("Content-Type")) {
      headers.set("Content-Type", "application/json");
    }
```